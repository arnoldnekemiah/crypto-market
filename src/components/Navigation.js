import { useState } from 'react';
import { useSelector } from 'react-redux';
import ClipLoader from 'react-spinners/ClipLoader';
import { NavLink } from 'react-router-dom';
import { filterByRankOrName } from './functions';

const Navigation = () => {
  const { cryptoArray, isLoading, error } = useSelector((store) => store.crypto);
  const [query, setQuery] = useState('');

  const filteredCrypto = query
    ? filterByRankOrName(cryptoArray, query)
    : cryptoArray; // Display all cryptocurrencies when query is empty

  if (isLoading) {
    return (
      <div id="loading" className="row">
        <ClipLoader
          id="loader"
          loading
          color="#5788e4"
          size="8rem"
          cssOverride={{
            borderWidth: '7px',
          }}
          speedMultiplier={1}
        />
      </div>
    );
  }

  if (error) {
    return <div id="errorMsg" className="row"><h1>{error}</h1></div>;
  }

  return (
    <nav id="coinNav">
      <form
        id="form"
        onSubmit={(e) => {
          e.preventDefault();
        }}
      >
        <input
          type="text"
          placeholder="Filter by Rank or Name"
          value={query}
          onChange={(e) => setQuery(e.target.value)}
        />
      </form>
      <ul id="coinGrid">
        {filteredCrypto.map((crypto) => (
          <li key={crypto.rank} className="gridItem">
            <NavLink className="link row gridLink" to={`/${crypto.rank}`}>
              <h2>{`#${crypto.rank}`}</h2>
              <p>{`${crypto.name}`}</p>
            </NavLink>
          </li>
        ))}
      </ul>
      {!filteredCrypto.length && query && (
        <div id="searchResult">
          <span id="searchError" className="column">
            Out Of Bounds
          </span>
        </div>
      )}
    </nav>
  );
};

export default Navigation;
