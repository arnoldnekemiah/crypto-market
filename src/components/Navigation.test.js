import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Navigation from './Navigation';

// Create a mock Redux store
const mockStore = configureStore([]);

describe('Navigation', () => {
  it('renders correctly', () => {
    const initialState = {
      crypto: {
        cryptoArray: [], // Replace with your mock data
        isLoading: false,
        error: null,
      },
    };

    const store = mockStore(initialState);

    const component = renderer.create(
      <Provider store={store}>
        <Navigation />
      </Provider>,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
