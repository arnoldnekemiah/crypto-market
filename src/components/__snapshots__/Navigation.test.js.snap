// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation renders correctly 1`] = `
<nav
  id="coinNav"
>
  <form
    id="form"
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Filter by Rank or Name"
      type="text"
      value=""
    />
  </form>
  <ul
    id="coinGrid"
  />
</nav>
`;
